zeppelin-plugins/launcher/k8s-standard/src/main/java/org/apache/zeppelin/interpreter/launcher/K8sRemoteInterpreterProcess.java

```diff
@@ -0,0 +1,379 @@
+package org.apache.zeppelin.interpreter.launcher;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.zeppelin.interpreter.remote.RemoteInterpreterProcess;
+import org.apache.zeppelin.interpreter.remote.RemoteInterpreterUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class K8sRemoteInterpreterProcess extends RemoteInterpreterProcess {
+  private static final Logger LOGGER = LoggerFactory.getLogger(K8sStandardInterpreterLauncher.class);
+  private static final int K8S_INTERPRETER_SERVICE_PORT = 12321;
+  private final Kubectl kubectl;
+  private final String interpreterGroupId;
+  private final String interpreterGroupName;
+  private final String interpreterSettingName;
+  private final File specTempaltes;
+  private final String containerImage;
+  private final Properties properties;
+  private final Map<String, String> envs;
+  private final String zeppelinServiceHost;
+  private final String zeppelinServiceRpcPort;
+  private final int podCreateTimeoutSec = 180;
+
+  private final Gson gson = new Gson();
+  private final String podName;
+  private final boolean portForward;
+  private final String sparkImage;
+  private ExecuteWatchdog portForwardWatchdog;
+  private int podPort = K8S_INTERPRETER_SERVICE_PORT;
+
+  private AtomicBoolean started = new AtomicBoolean(false);
+
+  public K8sRemoteInterpreterProcess(
+          Kubectl kubectl,
+          File specTemplates,
+          String containerImage,
+          String interpreterGroupId,
+          String interpreterGroupName,
+          String interpreterSettingName,
+          Properties properties,
+          Map<String, String> envs,
+          String zeppelinServiceHost,
+          String zeppelinServiceRpcPort,
+          boolean portForward,
+          String sparkImage,
+          int connectTimeout) {
+    super(connectTimeout);
+    this.kubectl = kubectl;
+    this.specTempaltes = specTemplates;
+    this.containerImage = containerImage;
+    this.interpreterGroupId = interpreterGroupId;
+    this.interpreterGroupName = interpreterGroupName;
+    this.interpreterSettingName = interpreterSettingName;
+    this.properties = properties;
+    this.envs = new HashMap(envs);
+    this.zeppelinServiceHost = zeppelinServiceHost;
+    this.zeppelinServiceRpcPort = zeppelinServiceRpcPort;
+    this.portForward = portForward;
+    this.sparkImage = sparkImage;
+    this.podName = interpreterGroupName.toLowerCase() + "-" + getRandomString(6);
```

Comment:
Is it better to use timestamp instead of RandomString?

Reply:
Could you explain a bit more why timestamp can be better?

Reply:
This was just a thought: string might collide sometimes while timestamp would be less prone for collision.

Reply:
While code generating 6 char long random string using lower case alphabet, probability of collision is `<number of running interpreter pods>/26^6` = `<number of running interpreter pods>/308,915,776`. I think probability of collision is small enough in practical. 

One worry using timestamp in pod name is, pod name becomes predictable and that makes accessing to spark-ui insecure, while spark-ui is accessed through subdomain which is constructed based on interpreter pod name.

Reply:
Agree. Let it be random string.